index.js

```diff
@@ -76,8 +78,13 @@ module.exports = function(options){
                 if(!shouldCreateSourcemaps) return fileToAddTo;
 
                 var map = result.maps[blessOutputIndex];
-                map.file = fileToAddTo.relative;
-                map.sources = [file.relative];
+                map.file = path.relative(fileToAddTo.base, fileToAddTo.path);
+                map.sources = [path.relative(file.base, file.path)];
+
+                //gotta assign fileToAddTo a source map so applyScourceMap can merge original
+                //and blessed sourcemap.
+                fileToAddTo.sourceMap = JSON.parse(JSON.stringify(file.sourceMap));
```

Comment:
This is to make a copy of `file.sourceMap` so that manipulation `fileToAddTo.sourceMap` will not affect `file.sourceMap`. If we don't do a copy then somewhere a long the pipeline `file.sourceMap` will get corrupted, which will result in some weird mapping file.